### ISSUE TITLE
No dependency provided for _ on _ - The underlying issue, provides meta data not including non-interface inherited types 

### ISSUE THREAD

**Issue by decebals** [avaje/avaje-inject#167]
I try the following scenario (two controllers - class that extends `Controller` - injected in application instance):
```java
/* Simplified class */
public class Controller {

    Map<String, String> getContext() {
        return new HashMap<>();
    }

}
```
```java
@Singleton
public class Controller1 extends Controller {

    @Override
    Map<String, String> getContext() {
        Map<String, String> context = super.getContext();
        context.put("something", "1");

        return context;
    }

}
```
```java
@Singleton
public class Controller2 extends Controller {

    @Override
    Map<String, String> getContext() {
        Map<String, String> context = super.getContext();
        context.put("something", "2");

        return context;
    }

}
```
```java
@Singleton
public class Application {

    private final List<Controller> controllers;

    @Inject
    public Application(List<Controller> controllers) {
        this.controllers = controllers;
    }

    public List<Controller> getControllers() {
        return controllers;
    }

}
```
```java
public class Test {

    public static void main(String[] args) {
        BeanScope beanScope =  BeanScope.newBuilder().build();
        Application application = beanScope.get(Application.class);
        List<? extends Controller> controllers = application.getControllers();
        System.out.println("controllers.size() = " + controllers.size());
    }

}
```

The result of running `Test` is:
```
java: No dependency provided for test.Controller on test.Application
```

I attached the test project.
[avaje-injector-test.zip](https://github.com/avaje/avaje-inject/files/7610384/avaje-injector-test.zip)

In Spring and Guice I don't have this problem.

Thanks!

**Comment by rbygrave** [comment]
This is a bug, the underlying issue is that the "meta data" for Controller1 and Controller2 is not including `Controller`.

That is, the `provides` meta data is only including interface types and is not including the inherited types (in this case `Controller`).

**Comment by rbygrave** [comment]
For some more detail on this issue.  When we look at the generated TestModule we see the meta data on the methods that create Controller1 and Controller2. We currently see:

```java
  ...
  @DependencyMeta(type="test.Controller1")
  protected void build_test_Controller1() {
    Controller1$DI.build(builder);
  }

  @DependencyMeta(type="test.Controller2")
  protected void build_test_Controller2() {
    Controller2$DI.build(builder);
  }

  @DependencyMeta(type="test.Application", dependsOn={"test.Controller"})
  protected void build_test_Application() {
    Application$DI.build(builder);
  }
```

And instead with the fix (version 5.16) we see the meta data for Controller1 and Controller2 with the correct `provides` as

```java

  @DependencyMeta(type="test.Controller2", provides={"test.Controller"})
  protected void build_test_Controller2() {
    Controller2$DI.build(builder);
  }

  @DependencyMeta(type="test.Controller1", provides={"test.Controller"})
  protected void build_test_Controller1() {
    Controller1$DI.build(builder);
  }

  @DependencyMeta(type="test.Application", dependsOn={"test.Controller"})
  protected void build_test_Application() {
    Application$DI.build(builder);
  }

```
